components/searchTodo.js

```diff
@@ -0,0 +1,28 @@
+import { searchTodoInput } from "./domElements.js";
+import { globalTodos, setTodos } from "../entry.js";
+import { renderTodoList } from "./domActions.js";
+
+function searchTodo() {
+  const searchedText = searchTodoInput.value.toLowerCase();
+
+  const searchedTodos = globalTodos.filter((todo) =>
+    todo.title.toLowerCase().includes(searchedText)
+  );
+
+  setTodos(searchedTodos);
+  renderTodoList();
+}
+
+export function debounceSearch(fn, delay) {
```

Comment:
We can add a default delay
```suggestion
export function debounceSearch(fn, delay = 500) {
```